### PR TITLE
web ui: rename folders #2886

### PR DESCRIPTION
**Pull request by raphi011**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by raphi011** [comment]
#2886

**Comment by markus2330** [comment]
Thank you very much for working on this!

File: doc/tutorials/install-webui.md [hunk 1]
```diff
@@ -27,12 +27,12 @@ To build Elektra with the elektra-web tool:
 
 - Start an elektrad instance: `kdb run-elektrad`
 - Start the client: `kdb run-web`
-- You can now access the client on: [http://localhost:33334](http://localhost:33334)
+- You can now access the client on: `http://localhost:33334`
```
Comment (markus2330) [inline doc/tutorials/install-webui.md]: Any specific reason why you do not like the links?

Comment (raphi011) [reply doc/tutorials/install-webui.md]: the linkchecker build step fails

Comment (markus2330) [reply doc/tutorials/install-webui.md]: localhost can be safely added as exception :wink:

Comment (raphi011) [reply doc/tutorials/install-webui.md]: didn't know its possible to add exceptions

Comment (markus2330) [reply doc/tutorials/install-webui.md]: tests/linkchecker.whitelist

Comment (raphi011) [reply doc/tutorials/install-webui.md]: it was easy to find as soon as i knew it existed :) fixed it

**Comment by raphi011** [comment]
@markus2330 are you ok with adding the 'client' (now webui) folder to the 'webd' folder? Doesn't really make a lot of sense to split them up since webd depends on (serves) the client.

**Comment by markus2330** [comment]
Would this make something simpler? From the build system I think keeping it separated is nice-to-have, as it allows to compile the individual parts. Also from development point of view it would be good to keep it separated, e.g. it might be useful to reimplement webd in a different language.

**Comment by raphi011** [comment]
> Would this make something simpler? From the build system I think keeping it separated is nice-to-have, as it allows to compile the individual parts. Also from development point of view it would be good to keep it separated, e.g. it might be useful to reimplement webd in a different language.

Well at the moment it just does not make sense to compile the client without webd or the other way around because webd serves the client bundle.

**Comment by markus2330** [comment]
Yes, for the compilation it might be not needed. You can define a single name that includes "both tools".

Merging the folders though seems to me like a lot of error-prone work and I do not see the benefit yet.

**Comment by raphi011** [comment]
> Yes, for the compilation it might be not needed. You can define a single name that includes "both tools".
> 
> Merging the folders though seems to me like a lot of error-prone work and I do not see the benefit yet.

good idea 'web' now builds all 3 tools: elektrad, webd and webui. It is still possible to compile them separetely.

**Comment by mpranj** [comment]
Are `src/tools/elektrad/elektrad` and `src/tools/elektrad/.dir/default.ecf` checked in for a reason?

**Comment by raphi011** [comment]
> Are `src/tools/elektrad/elektrad` and `src/tools/elektrad/.dir/default.ecf` checked in for a reason?

good catch! I removed them

**Comment by raphi011** [comment]
Something's wrong with the link checker, it keeps on failing

**Comment by mpranj** [comment]
> Something's wrong with the link checker, it keeps on failing

Looks like all links point to github.com. Is it possible they blocked cirrus due to too many requests/abuse? :confused:

**Comment by markus2330** [comment]
@mpranj where do you see that the links point to github.com?

I only saw that is also some weird garbage at the end of the output (������)

**Comment by mpranj** [comment]
It can be seen in the [build log](https://cirrus-ci.com/task/5756264810545152).
```
Broken Links:
—————————————
doc/tutorials/contributing-clion.md:260:0 https://github.com/ElektraInitiative/libelektra/compare
doc/news/2015-04-03_0.8.11.md:178:0 https://github.com/ElektraInitiative/libelektra/blob/master/src/include/kdbproposal.h
```

**Comment by raphi011** [comment]
Since the build succeeded before and I didn't change any links - can we merge this nonetheless?

**Comment by mpranj** [comment]
Sure, go ahead! Thank you!

**Pull request merged**